middleware.js

```diff
@@ -0,0 +1,30 @@
+import { getToken } from "next-auth/jwt";
+import { NextResponse } from "next/server";
+
+export async function middleware(req) {
+  const session = await getToken({
+    req: req,
+    secret: process.env.NEXTAUTH_SECRET,
+  });
+  const reqPathName = req.nextUrl.pathname;
+
+  if (!session) {
+    if (reqPathName.startsWith("/api")) {
+      return NextResponse.json({}, { status: 401 });
+    }
+    return NextResponse.redirect(new URL("/auth/signin", req.url));
+  }
+
+  const username = session.username;
+  if (!process.env.ADMIN_USERS.includes(username)) {
+    if (reqPathName.startsWith("/api")) {
+      return NextResponse.json({}, { status: 401 });
+    }
+    return NextResponse.redirect(new URL("/404", req.url));
+  }
+  req.cookies.set("usernames", username);
```

Comment:
Why is this needed?

Reply:
sorry this is not needed, will remove it. I was trying to set the username in cookies, so we can get it in the Component page without using getServerSession.